tests/utils/test_exporter.py

```diff
@@ -241,7 +435,7 @@ def test_exporter_can_export_requirements_txt_with_nested_packages_and_markers(
             "c==7.8.9; sys_platform == 'win32' and python_version < '3.7'"
         ),
         "d": dependency_from_pep_508(
-            "d==0.0.1; python_version < '3.7' and platform_system == 'Windows' and sys_platform == 'win32'"
+            "d==0.0.1; platform_system == 'Windows' and python_version < '3.7' or sys_platform == 'win32' and python_version < '3.7'"
```

Comment:
@abn This change changes tested behavior but I'd argue that the previous behavior was in fact wrong:

  * `a` has `python_version < 3.7`
  * `a` depends on `b` & `c` which have the markers `platform_system == 'Windows'` and `sys_platform == 'win32'` respectively
  * `b` and `c` both depends on `d` and as such their markers should trickle down, but `d` should get installed whenever *one* of the parent markers is true, not only when *both* are true.

On the other hand this `python_version<3.7` is required for all packages, so the constraint could be rewritten as:
```
(platform_system == 'Windows' or sys_platform == 'win32' ) and python_version < '3.7'"
```
but that would probably be more work (and the current form is technically correct). For large dependency trees it might be worth to get the python version out of the nested constraints as far as possible though. But I guess this might require plenty of specialcasing.